staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_dispatcher.go

```diff
@@ -331,7 +331,38 @@ func CollectParams(
 			return nil, fmt.Errorf("paramRef.name and paramRef.selector are mutually exclusive")
 		}
 
-		switch param, err := paramStore.Get(paramRef.Name); {
+		// Try to get the param from the cache. If it's not found, retry with backoff
+		// to handle the race condition where resources (ConfigMap, Policy, Binding, and Param)
+		// are created in quick succession. The informer cache may have completed its initial
+		// sync (checked by WaitForCacheSync above), but newly created params might not have
+		// propagated to the cache yet. This retry logic with exponential backoff gives the
+		// cache a chance to catch up, avoiding false "param not found" errors during batch
+		// resource creation scenarios
+		var param runtime.Object
+		var err error
+		const maxRetries = 3
+		retryDelay := 50 * time.Millisecond
```

Comment:
I'm concerned that, over time, there will be additional changes requests to increase this limit to better handle control plane configurations where the the cache is farther behind.

How about, instead, we wire in a client and make a direct get as a fallback if we have a cache miss?